### PR TITLE
Set `WM_CLASS` for some windows

### PR DESCRIPTION
**Pull request by slotThe**
### Description

Fixes:
  - https://github.com/xmonad/xmonad-contrib/issues/526
  - https://github.com/xmonad/xmonad-contrib/issues/369

Set the WM_CLASS property for long-living windows that need it.  This
helps users to properly match on these windows for certain tasks.

This needs a new X11 release, hence it's a draft for now.  I also didn't change the bounds in the cabal file, as #545 has already done this and I expect this will get merged sooner. 

I was a bit conservative with what to set properties for.  Things like X.U.EZConfig also create windows that may very well have such a property, but I didn't think they were long-lived enough to warrant it really.  Feel free to point me to more modules that need it and I'll add them.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've considered how to best test these changes (property, unit,
        manually, ...) and concluded: 

    I tested this by hand and it works.  I don't think we can really tests these kinds of interaction with the X server.

  - [x] I updated the `CHANGES.md` file

  ~~- [ ] I updated the `XMonad.Doc.Extending` file (if appropriate)~~

### CODE REVIEW

**Review by liskin — APPROVED** [review]
Looks okay, might be nice to mention the motivation in the changelog as well so that people know what they can use it for.

**Comment by slotThe** [comment]
Sure, I've rebased and mentioned this in the CHANGES

File: CHANGES.md [hunk 1]
```diff
@@ -27,6 +27,9 @@
       constraint (was: `IO`), due to changes in how the xmonad core handles XDG
       directories.
 
+    - The prompt window now sets a `WM_CLASS` property.  This allows to
+      match on it in a manageHook.
```
Comment (liskin) [inline CHANGES.md:31]: Seems to me you also forgot what the motivation was. xmonad wouldn't be stupid enough to try to manage its own windows would it? This was more about shadows and rounded corners in a compositor, I think.

Comment (slotThe) [reply CHANGES.md:31]: Oh :)  Very good point yes, I'll change that

**Comment by liskin** [comment]
Note that https://github.com/xmonad/xmonad-contrib/issues/526 and https://github.com/xmonad/xmonad-contrib/issues/369 as GitHub apparently doesn't understand your multiline trailers. Might want to consider using the standard git trailers instead.

**Comment by slotThe** [comment]
On Wed, Jun 02 2021 03:49, Tomáš Janoušek wrote:
> Note that https://github.com/xmonad/xmonad-contrib/issues/526 and
> https://github.com/xmonad/xmonad-contrib/issues/369 didn't get closed
> as GitHub apparently doesn't understand your multiline trailers. Might
> want to consider using the standard git trailers instead.

Oh good point, thanks.  I'll do that next time

**Pull request merged**